src/makedeb/functions/dependency_checks/install_missing_dependencies.sh

```diff
@@ -1,30 +1,33 @@
 install_missing_dependencies() {
-  if [[ "${#missing_dependencies}" == "0" && "${#missing_build_dependencies}" == "0" ]]; then
-    return 0
-  fi
-  
-  msg2 "Installing required build dependencies..."
-  
-  if [[ "${target_os}" == "debian" ]]; then
-    if ! sudo apt-get install "${apt_args[@]}" -- "${missing_dependencies[@]}" "${missing_build_dependencies[@]}"; then
-      error "There was an error installing build dependencies."
-      exit 1
+    if [[ "${#missing_dependencies}" == "0" && "${#missing_build_dependencies}" == "0" ]]; then
+        return 0
     fi
-    
-    if ! sudo apt-mark auto -- "${missing_dependencies_no_relations[@]}" "${missing_build_dependencies_no_relations[@]}" 1> /dev/null; then
-      error "There was an error marking installed build dependencies as automatically installed."
-      error "You may need to mark the following packages as automatically installed via 'apt-mark auto':"
-      
-      for i in "${missing_dependencies_no_relations[@]}" "${missing_build_dependencies_no_relations[@]}"; do
-        msg2 "${i}"
-      done
-      exit 1
+
+    msg2 "Installing required build dependencies..."
+
+    if [[ "${target_os}" == "debian" ]]; then
+        if ! sudo apt-get satisfy "${apt_args[@]}" -- "${missing_dependencies[@]}" "${missing_build_dependencies[@]}"; then
```

Comment:
This is an important change - we were using `apt-get install` previously, but using `apt-get satisfy` gives us more reliability into the versions of packages we install as we can specify relationship operators (i.e. `golang-go (>> 2)` instead of just `golang-go`).

This was reported to not work on Debian Buster (#15), but that's going to be EOL soon, and I think it's worth the trade off if the majority of makedeb users face a benefit by it.